neurom/features/morphology.py

```diff
@@ -71,26 +71,48 @@
 feature = partial(feature, namespace=NameSpace.NEURON)
 
 
-def _map_neurites(function, morph, neurite_type, use_subtrees=False):
+def _map_neurites(function, morph, neurite_type):
     return list(
         iter_neurites(
             obj=morph,
             mapfun=function,
             filt=is_type(neurite_type),
-            use_subtrees=use_subtrees,
         )
     )
 
 
-def _map_neurite_root_nodes(function, morph, neurite_type, use_subtrees=False):
-    neurites = iter_neurites(obj=morph, filt=is_type(neurite_type), use_subtrees=use_subtrees)
-    return [function(neurite.root_node) for neurite in neurites]
+def _map_neurite_root_nodes(function, morph, neurite_type):
+    neurite_type = SubtypeCollection(neurite_type)
+    neurites = list(iter_neurites(obj=morph, filt=is_type(neurite_type)))
+    if neurite_type != NeuriteType.all:
```

Comment:
Is it possible to check if the neurite type is a composite type instance?

Reply:
Instead of doing two passes it would be cleaner if the logic determined the correct filter and then do once the iter_neurites with that filter. sth like:

```python
if neurite_type == NeuriteType.all:
    filt = is_type(neurite_type)
else:
    filt = lambda neurite: SubtypeCollection(neurite.type).root_type == neurite_type

return [function(trunk.root_node) for trunk in iter_neurites(obj=morph, filt=filt)]
```

Reply:
Indeed it's better!